pkg/telemetry/telemetry.go

```diff
@@ -159,6 +160,17 @@ func (c Component) sendTelemetry(ctx context.Context) {
 	}
 }
 
+func (c Component) addCustomData(ctx context.Context, analyticCtx *analytics.Context) {
+	cm, err := c.kubernetesClient.CoreV1().ConfigMaps("kube-system").Get(ctx, "k0s-telemetry", metav1.GetOptions{})
+	if err != nil {
+		c.log.WithError(err).Info("can't get k0s-telemetry configmap, not adding custom data")
+		return
+	}
+	for k, v := range cm.Data {
+		analyticCtx.Extra[k] = v
```

Comment:
As the "extra" is a fake field and its contents are just serialized in the json, this makes it possible to overwrite existing data in the analytics json, accidentally or, maybe in some far fetched scenario, on purpose.

Maybe prefix the keys from the custom data with something like `custom.`?

Reply:
Hmm, I wonder if Extra could take the cm.Data as is, something like `analyticsCtx.Extra["custom"] = cm.Data`

Reply:
> Maybe prefix the keys from the custom data with something like custom.?

This sounds like a good idea to me 👍